src/main/scala/com/gu/contentapi/firehose/kinesis/EventProcessor.scala

```diff
@@ -0,0 +1,119 @@
+package com.gu.contentapi.firehose.kinesis
+
+import com.amazonaws.services.kinesis.model.Record
+import com.amazonaws.services.kinesis.clientlibrary.interfaces.{ IRecordProcessor, IRecordProcessorCheckpointer }
+import com.amazonaws.services.kinesis.clientlibrary.types.ShutdownReason
+import com.typesafe.scalalogging.LazyLogging
+import com.twitter.scrooge.ThriftStruct
+import java.util.concurrent.atomic.{ AtomicInteger, AtomicLong }
+import java.util.{ List => JList }
+
+import com.gu.contentapi.firehose.kinesis.EventProcessor.EventWithSize
+
+import scala.collection.JavaConverters._
+import scala.concurrent.duration._
+import scala.util.{ Failure, Success }
+
+abstract class EventProcessor[EventT <: ThriftStruct](
+  checkpointInterval: Duration = 30.seconds,
+  maxCheckpointBatchSize: Int = 20
+)
+    extends IRecordProcessor
+    with ThriftDeserializer[EventT]
+    with LazyLogging {
+
+  private[this] var shardId: String = _
+
+  /* Use atomic to prevent any concurrent access issues */
+  private[this] val lastCheckpointedAt = new AtomicLong(System.nanoTime())
+  private[this] val recordsProcessedSinceCheckpoint = new AtomicInteger()
+
+  override def initialize(shardId: String): Unit = {
+    this.shardId = shardId
+    logger.info(s"Initialized an event processor for shard $shardId")
+  }
+
+  override def processRecords(records: JList[Record], checkpointer: IRecordProcessorCheckpointer): Unit = {
+    processRecordsIfActivated(records, checkpointer)
```

Comment:
This delegation is a bit pointless. Let's get rid of the `processRecordsIfActivated` method.